### ISSUE TITLE
crash in message_add

### ISSUE THREAD

**Issue by angband-import** [angband/angband#933]
**Reported by peter.wiersig@gmail.com on 15 Sep 2007 22:29 UTC**
I hit a software bug earlier this day and recompiled svn [6603c09] (SVN r560) with debug information.

message_add (str=0x7fff8116b000 "The Large yellow snake crushes you.", type=41) at z-msg.c:113
113                     messages->tail = m->prev;
(gdb) bt
#0  message_add (str=0x7fff8116b000 "The Large yellow snake crushes you.", type=41) at z-msg.c:113
#1  0x0000000000473bb3 in msg_print_aux (type=41, msg=0x7fff8116b000 "The Large yellow snake crushes you.") at util.c:1451
#2  0x0000000000473cca in message_format (message_type=14595, extra=-20445, fmt=<value optimized out>) at util.c:1585
#3  0x0000000000444dc0 in make_attack_normal (m_idx=43) at melee1.c:443
#4  0x000000000044aeb7 in process_monsters (minimum_energy=100 'd') at melee2.c:3776
#5  0x000000000042a94a in play_game () at dungeon.c:1661
#6  0x0000000000480cd1 in main (argc=3, argv=0x7fff8116c2f0) at main.c:466
(gdb) p *messages
$3 = {head = 0x7f38b0, tail = 0x0, colors = 0x79b260, count = 2049, max = 2048}

I read a "Summon monster" scroll on my last turn.

**Comment by angband-import** [comment]
**Comment by peter.wiersig\@gmail.com on 15 Sep 2007 22:30 UTC**

```
(gdb) bt
#0  message_add (str=0x7fff8116b000 "The Large yellow snake crushes you.", type=41) at z-msg.c:113
#1  0x0000000000473bb3 in msg_print_aux (type=41, msg=0x7fff8116b000 "The Large yellow snake crushes you.")
    at util.c:1451
#2  0x0000000000473cca in message_format (message_type=14595, extra=-20445, fmt=<value optimized out>)
    at util.c:1585
#3  0x0000000000444dc0 in make_attack_normal (m_idx=43) at melee1.c:443
#4  0x000000000044aeb7 in process_monsters (minimum_energy=100 'd') at melee2.c:3776
#5  0x000000000042a94a in play_game () at dungeon.c:1661
#6  0x0000000000480cd1 in main (argc=3, argv=0x7fff8116c2f0) at main.c:466
(gdb) p *messages
$3 = {head = 0x7f38b0, tail = 0x0, colors = 0x79b260, count = 2049, max = 2048}

```

**Comment by angband-import** [comment]
**Comment by anonymous on 16 Sep 2007 17:21 UTC**
You should consider compiling the daily builds in debug by default.
I hit a crash today too, but I'm not if I can provide useful info without symbols

**Comment by angband-import** [comment]
**Modified by takkaria on 21 Sep 2007 14:20 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 21 Sep 2007 14:38 UTC**
Testcases would have covered this, but since I dont have time to produce those right now, there should be a fix in [178c9a5] (SVN r574).

**Comment by angband-import** [comment]
**Modified by takkaria on 21 Sep 2007 14:38 UTC**
